lib/pos_consensus.go

```diff
@@ -766,7 +766,8 @@ func (fc *FastHotStuffConsensus) fetchValidatorListsForSafeBlocks(blocks []*MsgD
 
 	// Create a UtxoView for the committed tip block. We will use this to fetch the validator set for
 	// all of the safe blocks.
-	utxoView, err := NewUtxoView(fc.blockchain.db, fc.params, fc.blockchain.postgres, fc.blockchain.snapshot, nil)
+	utxoView, err := NewUtxoViewWithSnapshotCache(
+		fc.blockchain.db, fc.params, fc.blockchain.postgres, fc.blockchain.snapshot, nil, fc.blockchain.snapshotCache)
```

Comment:
Seems like you could use GetcommittedTipView() here too? Not sure what the eventManager would be used for internally within the UtxoView since the resulting view is only used to build a view of the blockchain state at for a particular block hash
```suggestion
	utxoView, err := bc.GetCommittedTipView()
```